File: my-folder/0506-relative-ranks/solution.cpp
class Solution {
public:
    vector<string> findRelativeRanks(vector<int>& score) {

        int n = score.size();
        vector<string>ans(n);

        map<int, int>m;


        for(int i=0; i<score.size(); i++){
            m[score[i]] = i;
        }
        
        priority_queue<int>pq(score.begin(), score.end());

        int x = 1;

        while(!pq.empty()) {
            int mx = pq.top(); pq.pop();
            if(x == 1) ans[m[mx]] = "Gold Medal";
            else if(x == 2) ans[m[mx]] = "Silver Medal";
            else if(x == 3) ans[m[mx]] = "Bronze Medal";

            else {
                ans[m[mx]] = to_string(x);
            }

            x++;
        }
        return ans;
    }
};

File: my-folder/0506-relative-ranks/solution.py
class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:
        n=len(score)
        ans = [" "]*n
        d = defaultdict(int)
        for i, k in enumerate(score):
            d[k] = i

        score.sort(reverse=True)
     
        for i in range(n):
            if i==0:
                ans[d[score[i]]] = "Gold Medal"
            elif i == 1:
                ans[d[score[i]]] = "Silver Medal"
            elif i==2: 
                ans[d[score[i]]] = "Bronze Medal"
            else:
                ans[d[score[i]]] = str(i+1)
            
        return ans
        
